src/main/java/step2/Operator.java

```diff
@@ -0,0 +1,35 @@
+package step2;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum Operator {
+
+    PLUS("+", (firstNumber, secondNumber) -> firstNumber + secondNumber),
+    MINUS("-", (firstNumber, secondNumber) -> firstNumber - secondNumber),
+    MULTIPLY_BY("*", (firstNumber, secondNumber) -> firstNumber * secondNumber),
+    DIVIDED_BY("/", (firstNumber, secondNumber) -> {
+        if (secondNumber == 0) {
+            throw new IllegalArgumentException(CommonConstant.INVALID_DIVIDED_BY_ZERO);
+        }
+        return firstNumber / secondNumber;
+    });
+
+    private String operator;
+    private Calculation calculation;
+
+    Operator(String operator, Calculation calculation) {
+        this.operator = operator;
+        this.calculation = calculation;
+    }
+
+    public static Operator of(String value) {
+        Optional<Operator> operators = Arrays.stream(values()).filter(o -> o.operator.equals(value)).findFirst();
+        return operators.orElseThrow(() -> new IllegalArgumentException(value + CommonConstant.INVALID_OPERATOR));
```

Comment:
개인적으로 메서드 체이닝 사용시 가독성을 위해 한줄에 하나의 dot(.)만 사용하는게 좋다고 생각해요.
또한 해당 로직들은 합쳐도 될 것 같아요 :)
```suggestion
        return Arrays.stream(values())
                     .filter(o -> o.operator.equals(value))
                     .findFirst()
                     .orElseThrow(() -> new IllegalArgumentException(value + CommonConstant.INVALID_OPERATOR));
```

Reply:
몇가지 규칙들은 알고있지만 막상 코딩하다보면 빠트리게되는거 같아요,,ㅋㅋㅋ
반영하겠습니다 !
람다와 스트림의 능숙한 활용이 더 필요할거 같아요 !